### PR TITLE
Use calendar.timegm when calculating cookie expiration

### PR DESCRIPTION
**Pull request by sigmavirus24**
Fixes #1859

Credit: @lukasa

### CODE REVIEW

**Comment by Lukasa** [comment]
We need to confirm that this will match the actual parsing logic of cookies before we merge this: we don't want to start messing up cookies created in this way.

**Comment by sigmavirus24** [comment]
That's a good point @Lukasa. I'm guessing @gazpachoking might have a good idea. I'm just too tired to think this hard right now =P

**Comment by Lukasa** [comment]
Me too, I'll take a look at this tomorrow.

**Comment by kennethreitz** [comment]
Any update?

**Comment by sigmavirus24** [comment]
Sorry I never got around to that. @Lukasa how about you?

**Comment by Lukasa** [comment]
Uh, no, totally dropped the ball here. I'll put it on my list. =)

**Comment by sigmavirus24** [comment]
@Lukasa how should we make sure this doesn't mess anything up?

**Comment by Lukasa** [comment]
I'm not sure, to be honest. I'm wondering if there's a good way we can confirm our cookie-parsing code works in all locales.

**Comment by kennethreitz** [comment]
Closing due to inactivity.

**Comment by Lukasa** [comment]
Ok, I'm reopening this because I've finally validated that the stdlib does this. That suggests that it's the right thing to do. I'll have to do the merge manually, but I'd like to have this.

@sigmavirus24, one question: could we meaningfully add this to a 2.9.1, or should it wait for 2.10.0?

**Comment by sigmavirus24** [comment]
I think this could be included in 2.9.1

File: requests/cookies.py [hunk 1]
```diff
@@ -393,8 +394,8 @@ def morsel_to_cookie(morsel):
         expires = time.time() + morsel['max-age']
     elif morsel['expires']:
         time_template = '%a, %d-%b-%Y %H:%M:%S GMT'
```
Comment (sigmavirus24) [inline requests/cookies.py:396]: Should the template have "GMT" in it?

Comment (Lukasa) [inline requests/cookies.py:396]: The honest answer is "probably not", to be honest: the RFC really doesn't require it at all, and instead only specifies that the time is to be parsed as UTC.

**Comment by Lukasa** [comment]
\o/ Finally fixed, nearly two years after the original patch was proposed!

**Pull request merged**